### ISSUE TITLE
Steam Account Linked to Multiple Forum Accounts 

### ISSUE THREAD

**Issue by hanman5000** [Lavoaster/IPS-SteamLoginMethod#70]
I think this is worth reporting, my sincere apologies if it is not.

Right now, you can "sign in with steam" AND add that same steam account to an existing account on the IPS website. 

If you do this it breaks the "sign in through steam" functionality (because I'm assuming it tries to log into both accounts, the existing board account + the steam created account). 

This is problematic because that user then cannot get into the account created via steam. 

Possible solution:
Make it so that if a user tries to link their steam account to their existing board account and they have already "signed in through steam" and created another board account an error message appears saying "Cannot link this steam account to this board account" or something along those lines.

Thanks!

**Comment by madman32395** [comment]
Moving forward, best would be to have the column set to Unique. But the issue originally lies on the end user creating multiple accounts.

I would be able to poke at this after holiday. RL job is utilizing me currently.

**Comment by kanalumaddela** [comment]
I didn't know this was an issue, but it's good to keep track of on our forums. My question is (since I don't want to start another topic). What if there was an efficient way to prevent multiple accounts being created?

Like a user signs up normally with email and password, but then another time tries to login with steam and then creates another account. What if when they try to do that, it checks the IP they register on against other users and if it matches a user, it'll ask them if that is their account. If so, they provide the login details and it'll associate that account with the steam account they logged in as.

We've dealt with that a lot quite recently, and end up doing a lot of merging.

**Comment by sejbr** [comment]
@hanman5000 @kanalumaddela
Not sure if you're still interested. But I did some changes on my version to fix those problems. If @madman32395 wishes, I could PR them to the repo. But I'm not sure if the problems I've experienced were somewhat intentional or just on my end.

First I've changed steamid column to unique.
Secondly, since the "cancel" didn't seem to remove the account which would lead to some problems with other users syncing the same accounts, I've added a hook to `\IPS\core\modules\front\system\register`
Inserting 
```php
if(\IPS\Member::loggedIn()->member_id && \IPS\Member::loggedIn()->steamid) {
	$member = \IPS\Member::load( \IPS\Member::loggedIn()->member_id );
	$member->delete(TRUE);
}
```

Between
```php
/* Member sync callback */
\IPS\Member::loggedIn()->memberSync( 'onLogout', array( \IPS\Http\Url::internal( '' ) ) );
		
[...]
		
/* Login handler callback */
foreach ( \IPS\Login::handlers( TRUE ) as $k => $handler )
```

Next, since It was possible to sync multiple accounts from user settings, I've added this to the `system/login/steam.php`
```php
if(\IPS\Db::i()->select('*', 'core_members', array('steamid=?', $steamId))->count()) {
	$lang = \IPS\Member::loggedIn()->language();
	\IPS\Output::i()->redirect( isset( $_SERVER['HTTP_REFERER'] ) ? $_SERVER['HTTP_REFERER'] : \IPS\Http\Url::internal( '' ), $lang->get('login_steamid_in_use'), 404 );
}
```

Between
```php
	$newMember = true;
	}
}

[...]

/* Create member */

$member->steamid = $steamId;
$member->save();
```

For that, I've also added a lang string to the lang.php array.
`'login_steamid_in_use' => 'SteamID already in use by another account!',`

Not sure if those were all changes I've made since I had to read a lot of IPB source, but I think this solves the biggest issues.
It might have been done better but I don't have enough time to read the whole IPB source.
Hope it helps you in any way.

**Comment by Lavoaster** [comment]
This was resolved in the update made to support 4.3.
